### PR TITLE
begin TS conversion

### PR DESCRIPTION
**Pull request by FezVrasta**
This is the first step in order to address #67

Let's convert the codebase to TypeScript, this way it will be much easier to migrate it from class components to hooks later, fixing #67.

I'm not sure I will have enough bandwidth to allocate to this effort so we may want to have you Kitware folks work on it if possible, we can sync offline.

### CODE REVIEW

File: src/core/Algorithm.tsx [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
+   */
+  port?: number;
+  /**
+   * vtkClass name
+   */
+  vtkClass?: string;
+  /**
+   * set of property values for vtkClass
+   */
+  state?: object;
```
Comment (FezVrasta) [inline src/core/Algorithm.tsx:24]: I used a codemod to convert prop types to TS types, they will need some improvemements.

**Comment by nyoung697** [comment]
I support this endeavour, and may have some time to contribute.

File: src/representations/VolumeDataRepresentation.tsx [hunk 2]
```diff
@@ -1,12 +1,78 @@
...
+   */
+  scalars?: number[];
+  /**
+   * Types of numbers provided in scalars
+   */
+  scalarsType?: string;
+  /**
+   * Properties to set to the mapper
+   */
+  mapper?: object;
```
Comment (nyoung697) [inline src/representations/VolumeDataRepresentation.tsx:45]: Do these objects map directly to classes like `vtkMapper`, `vtkVolume`, `vtkProperty`, etc...? Could the object eventually be converted to the types on those classes?

Comment (FezVrasta) [reply src/representations/VolumeDataRepresentation.tsx:45]: I'm not sure, they were defined just as Object on prop-types 😔

Comment (floryst) [reply src/representations/VolumeDataRepresentation.tsx:45]: These are objects that represent the settable properties for the mapper/property/etc. We can feasibly use the initial values types for the object constructors here. For instance, here is the [vtkMapper initial values interface](https://github.com/Kitware/vtk-js/blob/master/Sources/Rendering/Core/Mapper/index.d.ts#L26).

**Comment by floryst** [comment]
After some evaluation, this will additionally require a significant chunk of typing for certain classes in vtk.js (i.e. manipulators). This is a good thing, but will incur further overhead when making this migration.

**Pull request closed without merging**